code-challenges.test.js

```diff
@@ -23,32 +30,122 @@ var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saf
+    Input->an array
+    Output->an array containing the minimum and maximum values of the original array
+        create a function
+        declare maximum value to negative infinity
+        declare minimum value to positive infiity
+        loop the array
+        if array at i is greater than the declared maximum value, maximum equals array at i
+        if array at i is less than the declared minimum value ,minimum equals array at i
+        return an array with minimum and maximum values respectively 
+
+*/
+const minMax=(array)=>{
+    let max=-Infinity
+    let min=Infinity
+    for(let i=0;i<array.length;i++){
+        if(array[i]>max){
+            max=array[i]
+        }
+        if(array[i]<min){
+            min=array[i]
+        }
+    }return [min,max]
+}
+
+// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. 
+//STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 // a) Create a test with an expect statement using the variables provided.
-
+describe("removeDup",()=>{
+    test("merge the two arrays and remove duplicates",()=>{
+        expect(removeDup([3, 7, 10, 5, 4, 3, 3],[7, 8, 2, 3, 1, 5, 4])).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
+    })
+})
 var testArray1 = [3, 7, 10, 5, 4, 3, 3]
 var testArray2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
-
-
 // b) Create the function that makes the test pass.
+/*
+Algorithm->merge two arrays and remove duplicates 
+    Input->two array with duplicated values
+    Output->combined array with unique values
+        create a function
+        concatnate the two arrays
+        declare a variable and assign an empty array
+        loop over the merged array
+        if the declared variable doesn't contain elements of the merged array push to it
+        return the declared variable
+*/
+const removeDup=(array1,array2)=>{
+    let mergedArray=array1.concat(array2)
+    let unique=[]
+    for(let i=0;i<mergedArray.length;i++){
+        if(!unique.includes(mergedArray[i])){
+            unique.push(mergedArray[i])
+        }
+    }return unique
+}
+
+//STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+describe("removeDuplicate",()=>{
+    test("returns unique values from dynamic number of arguments",()=>{
+        expect(removeDuplicate([3,2,1],[1,2,3,4])).toEqual([1,2,3,4])
+    })
+    test("returns unique values from dynamic number of arguments",()=>{
+        expect(removeDuplicate([3,2,1],[1,2,3,4],[100,200,300,100])).toEqual([1,2,3,4,100,200,300])
+    })
+})
+const removeDuplicate=(...array)=>{
+    let merged=array.flat()
+     let unique=[]
+     for(let i=0;i<merged.length;i++){
+         if(!unique.includes(merged[i])){
+             unique.push(merged[i])
+         }
+     }return unique.sort((a,b)=>a-b)
+ }
+
+//test failed for question number 1 as I expected ,because the answer is 
+//randomly  ordered array
```

Comment:
Make sure you are testing as you code. Try to use the tools available to you to get those tests to pass. This is an important part of Quality Assurance.